lib/commands/device/startservice.ts

```diff
@@ -0,0 +1,155 @@
+/**
+ * @license
+ * Copyright 2016-2020 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Flags, Args } from '@oclif/core';
+import Command from '../../command';
+import * as cf from '../../utils/common-flags';
+import { getBalenaSdk, getCliUx, stripIndent } from '../../utils/lazy';
+import type {
+	BalenaSDK,
+	DeviceWithServiceDetails,
+	CurrentServiceWithCommit,
+} from 'balena-sdk';
+
+export default class DeviceStartServiceCmd extends Command {
+	public static description = stripIndent`
+		Start containers on a device.
+
+		Start containers on a device.
+		If the --service flag is provided, then only those services' containers
+		will be started.
+
+		Multiple devices and services may be specified with a comma-separated list
+		of values (no spaces).
+		`;
+	public static examples = [
+		'$ balena device startservice 23c73a1 --service myService',
+		'$ balena device startservice 23c73a1 -s myService1,myService2',
+	];
+
+	public static args = {
+		uuid: Args.string({
+			description:
+				'comma-separated list (no blank spaces) of device UUIDs to start',
+			required: true,
+		}),
+	};
+
+	public static usage = 'device startservice <uuid>';
+
+	public static flags = {
+		service: Flags.string({
+			description:
+				'comma-separated list (no blank spaces) of service names to start',
+			char: 's',
+		}),
+		help: cf.help,
+	};
+
+	public static authenticated = true;
+
+	public async run() {
+		const { args: params, flags: options } = await this.parse(
+			DeviceStartServiceCmd,
+		);
+
+		const balena = getBalenaSdk();
+		const ux = getCliUx();
+
+		const deviceUuids = params.uuid.split(',');
+		const serviceNames = options.service?.split(',');
+
+		// Iterate sequentially through deviceUuids.
+		// We may later want to add a batching feature,
+		// so that n devices are processed in parallel
```

Comment:
+1
I have a PR from a while ago that aims to begin adding that support here: https://github.com/balena-io/pinejs/pull/675
Haven't had the time to pushed it to the end, hopefully will find the time soon

Related roadmap item in case you are interested: https://roadmap.balena.io/posts/19/allow-changes-to-variables-to-be-batched-and-submitted-all-at-once

Reply:
Nice. Thanks for the information. Good to get a feel about the roadmap ahead.